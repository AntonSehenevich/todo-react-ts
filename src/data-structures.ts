export enum TodoFilter {
  All = 1,
  Completed,
  Uncompleted
}

export type Todo = {
  id: string
  text: string
  completed: boolean
}

export type TodoList = Todo[]

export type EditTodoParams = Pick<Todo, 'id' | 'text'>

export type ReorderTodosParams = {
  source: string
  destination: string
}
