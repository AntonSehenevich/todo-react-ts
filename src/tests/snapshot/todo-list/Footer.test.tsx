import renderer from 'react-test-renderer'
import Footer from '../../../components/todos/footer'
import { TodoFilter } from '../../../data-structures'

test('renders correctly', () => {
  const tree = renderer
    .create(
      <Footer
        filter={TodoFilter.All}
        uncompletedCount={1}
        deleteCompletedTodos={jest.fn()}
        setFilter={jest.fn()}
      />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
