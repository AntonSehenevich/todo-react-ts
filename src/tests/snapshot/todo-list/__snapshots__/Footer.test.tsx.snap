// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div>
    <span>
      1
       
      item
       left
    </span>
  </div>
  <div>
    <ul
      className="list"
      data-testid="todos-filter"
    >
      <li>
        <button
          className="button active"
          onClick={[Function]}
          type="button"
        >
          All
        </button>
      </li>
      <li>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          Active
        </button>
      </li>
      <li>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          Completed
        </button>
      </li>
    </ul>
  </div>
  <div>
    <button
      className="delete-completed-button"
      onClick={[MockFunction]}
      type="button"
    >
      Clear completed
    </button>
  </div>
</div>
`;
